### PR TITLE
[bugfix](fe) add check for leg/lead function params

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
leed/lag function
param 2 is const positive value
param 3 is const value 

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/Lead.java [hunk 1]
```diff
@@ -90,9 +91,13 @@ public void checkLegalityBeforeTypeCoercion() {
             return;
         }
         if (children().size() >= 2) {
-            DataType offsetType = getOffset().getDataType();
-            if (!offsetType.isNumericType()) {
-                throw new AnalysisException("The offset of LEAD must be a number:" + this.toSql());
+            checkValidParams(child(1));
+            if (((Literal) getOffset()).getDouble() <= 0) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/Lead.java:95]: checkValidParams not ensure param is literal, so there maybe throw cast error.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/Lead.java [hunk 2]
```diff
@@ -90,9 +91,13 @@ public void checkLegalityBeforeTypeCoercion() {
             return;
         }
         if (children().size() >= 2) {
-            DataType offsetType = getOffset().getDataType();
-            if (!offsetType.isNumericType()) {
-                throw new AnalysisException("The offset of LEAD must be a number:" + this.toSql());
+            checkValidParams(child(1));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/Lead.java:94]: it is better to call getOffset here

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/WindowFunction.java [hunk 3]
```diff
@@ -46,11 +48,25 @@ public boolean equals(Object o) {
...
 
     @Override
     public int hashCode() {
         return Objects.hash(getName(), children);
     }
+
+    /**
+     * LAG/LEAD param must be const number
+     */
+    public void checkValidParams(Expression param) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/WindowFunction.java:62]: protected is better

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**